confluent_kafka/schema_registry/avro.py

```diff
@@ -0,0 +1,329 @@
+    |                       | Instructs the AvroSerializer on how to Construct |
+    | subject.name.strategy | Schema Registry subject names.                   |
+    |                       | Defaults to topic_subject_name_strategy.         |
+    +-----------------------+--------------------------------------------------+
+
+    Schemas are registered to namespaces known as Subjects which define how a
+    schema may evolve over time. By default the subject name is formed by
+    concatenating the topic name with the message field separated by a hyphen.
+
+    i.e. {topic name}-{message field}
+
+    Alternative naming strategies may be configured with the property
+    `subject.name.strategy`.
+
+    Supported subject name strategies:
+    +--------------------------------------+------------------------------+
+    | Subject Name Strategy                | Output Format                |
+    +======================================+==============================+
+    | topic_subject_name_strategy(default) | {topic name}-{message field} |
+    +--------------------------------------+------------------------------+
+    | topic_record_subject_name_strategy   | {topic name}-{record name}   |
+    +--------------------------------------+------------------------------+
+    | record_subject_name_strategy         | {record name}                |
+    +--------------------------------------+------------------------------+
+
+    See ``Subject name strategy`` for additional details.
+
+    Note:
+        Prior to serialization all ``Complex Types`` must first be converted to
+        a dictionary instance. This may handled manually prior to calling
+        :py:func:`SerializingProducer.produce()` or by registering a `to_dict`
+        callable with the AvroSerializer.
+
+        See ``avroproducer.py`` in the examples directory for more details on
+        usage.
+
+    Args:
+        schema_registry_client (SchemaRegistryClient): Schema Registry client.
+
+        schema_str (str): Avro Schema declaration.
+
+        to_dict (callable): Callable(object) -> dict. Converts object to a dict.
+
+        conf (dict): AvroSerializer configuration.
+
+    .. _Subject name strategy:
+        https://docs.confluent.io/current/schema-registry/serializer-formatter.html#subject-name-strategy
+
+    .. _Schema declaration:
+        https://avro.apache.org/docs/current/spec.html#schemas
+
+    """  # noqa: E501
+    __slots__ = ['_hash', 'auto_register', 'known_subjects', 'parsed_schema',
+                 'registry', 'schema', 'schema_id', 'schema_name',
+                 'subject_name_func', 'to_dict']
+
+    # default configuration
+    _default = {'auto.register.schemas': True,
+                'subject.name.strategy': topic_subject_name_strategy}
+
+    def __init__(self, schema_registry_client, schema_str,
+                 to_dict=None, conf=None):
+        self.registry = schema_registry_client
```

Comment:
i'd make all these instance variables private. maybe `schema` is useful to expose... but even with that, probably err on making it private i think.